src/mminou.h

```diff
@@ -93,29 +102,270 @@ extern flag g_commandFileSilentFlag; /* For SUBMIT ... /SILENT */
+ *     user request (\ref g_quitPrint = 1), output is redirected to a string
+ *     (\ref g_outputToString = 1), \ref backFromCmdInput = 1
+ *     (\ref cmdInput uses only the scrolling features)
+ *
+ *      __todo__ clarify variants/invariants
+ *
+ * -# Print the prepared output onto the screen.
+ *
+ *     Some contexts prevent this step: Step (6) (line wrapping) was executed,
+ *     output is discarded on user request (\ref g_quitPrint = 1)
+ *     output is redirected to a string (\ref g_outputToString = 1), a SUBMIT
+ *     is silently executing (\ref g_commandFileSilentFlag = 1),
+ *     \ref backFromCmdInput = 1 (\ref cmdInput uses only the scrolling
+ *     features)
+ *
+ *     \ref printedLines is increased if the prepared output terminates with
+ *     LF.
+ *
+ * -# Add a new history page in \ref backBuffer for the output generated in
+ *     step (4) if the current page has overflown.  Do this even when scrolling
+ *     is disabled.
+ *
+ *     Some contexts prevent this step: Step (7) was not executed,
+ *     \ref printedLines <= \ref g_screenHeight allows another line of output,
+ *     the output printed in step (7) has no trailing LF and is appended to the
+ *     last line of output, a SUBMIT is silently executing
+ *     (\ref g_commandFileSilentFlag = 1), scrolling is enabled
+ *     (\ref g_scrollMode = 1, step (3) handles this case).
+ *
+ *     \ref printedLines is set to 1.  \ref backBufferPos is increased.
+ *     \ref g_pntrTempAllocStackTop may be reset to
+ *     \ref g_pntrStartTempAllocStack.
+ *
+ * -# Add the output to the \ref backBuffer.
+ *
+ *     Some contexts prevent this step: Step (7) was not executed.
+ *
+ *     \ref g_tempAllocStackPos is reset to \ref g_startTempAllocStack.
+ *
+ * -# Log the output to a file given by \ref g_logFilePtr.
+ *
+ *     Some contexts prevent this step: Step (4) was not executed, a log file
+ *     is not opened (\ref g_logFileOpenFlag = 0), output is redirected to a string
+ *     (\ref g_outputToString = 1), \ref backFromCmdInput = 1 (\ref cmdInput
+ *     uses only the scrolling features)
+ *
+ *     \ref g_tempAllocStackPos is reset to \ref g_startTempAllocStack.
+ *
+ * -# Copy the prepared output in step (4) to \ref g_printString.
+ *
+ *     Some contexts prevent this step: Step (4) was not executed, output is
+ *     not redirected to a string (\ref g_outputToString = 0),
+ *     \ref backFromCmdInput = 1 (\ref cmdInput uses only the scrolling
+ *     features)
+ * .
+ *
+ * \param[in] fmt NUL-terminated text to display with embedded placeholders
+ *   for insertion of data (which are converted into text if necessary) pointed
+ *   to by the following parameters.  The placeholders are encoded in a cryptic
+ *   syntax explained
+ *   <a href="https://en.wikipedia.org/wiki/Printf_format_string">here</a> or
+ *   <a href="https://en.cppreference.com/w/c/io/fprintf">here</a>.
+ *   This parameter is ignored when \ref backFromCmdInput is 1.
```

Comment:
When reading this text I thought it was going to link to some page on metamath custom syntax. I would go for something a little more neutral than "cryptic syntax":
```suggestion
 * \param[in] fmt NUL-terminated text to display with embedded placeholders
 *   for insertion of data (which are converted into text if necessary) pointed
 *   to by the following parameters.  The format string uses the same syntax as
 *   [printf](https://en.cppreference.com/w/c/io/fprintf).
 *   This parameter is ignored when \ref backFromCmdInput is 1.
```